dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java

```diff
@@ -1782,6 +1784,11 @@ private void submitStandByTask() {
                     TaskInstance taskInstance = submitTaskExec(task);
                     if (taskInstance == null) {
                         this.taskFailedSubmit = true;
+                        // Remove and add to complete map and error map
+                        removeTaskFromStandbyList(task);
```

Comment:
It's better to retry if task submit fail, but not remove and mark it error and complete. What do you think?

Reply:
The main possible reason for the submit fail is that the submit task to db error, and has been retried, so I don't think it can be tried again. What do you think?

Reply:
OK, I had changed my mind and agree with you. Due to it already has retry submit logic, so if task submit fail finally, I think mark it error is OK.